### ISSUE TITLE
Consider supplying command to check for mixed conda-forge and defaults in an environment

### ISSUE THREAD

**Issue by ceball** [pyviz-dev/pyct#33]
Binary incompatibility is so often a cause of problems (and often not obvious at first), and mixing channels by mistake is easy even when trying not to. Supplying a command that checks an environment is reasonably sane might be helpful. E.g. `earthsim.check_env()`, `earthsim check-env` (or similar).

At the moment, without such a command, someone can always just run conda list and send us the output to inspect. But a command would let them get an indication without asking us. And we ourselves appear to repeatedly fall victim to problems too...

We could go further, maybe running the command automatically on import for some frequently troubled packages and warning (e.g. `import pyviz` might lead to `Warning: current environment mixes conda-forge and defaults...`).

Or we could just detect the commonest problems, e.g. gdal, scipy/deps, etc.

Not sure exactly what I'm proposing here yet.

**Comment by ceball** [comment]
Some more info in https://github.com/pyviz/EarthSim/issues/104#issuecomment-392843921 and subsequent comments.

**Comment by ceball** [comment]
I think this issue should be renamed, split up, reworded, repurposed, thought about some more, closed, or something like that :)

I see two bits in the original request

### report installation source

Reporting installation source would be useful, if that information is available. E.g. by conda, which channel, etc.

There's overlap with what package managers like conda already do, of course. 

pyct version reporting is nice because you can import and use in your current running process, where you are having the problem you're reporting - there's no confusion about getting the environment details at a different time or place (e.g. running later at command prompt).

### checking more complex stuff like binary compatibility

I believe the immediate problem of extremely frequent incompatibility between c-f and main has gone away. So probably the motivation for this aspect is much less strong now. Of course incompatibilities can (and do) still exist - if pursuing, maybe we'd just consider adding some support for this for common problems we encounter in holoviz? Or we could just add something to report, "you're using linux, and mixing multiple conda channels plus pypi; if you're thinking of reporting some obscure geo stack problem to us, please don't ;)"

**Comment by jbednar** [comment]
It's been some time since I've encountered binary conflicts between conda-forge and defaults. It's true I was on a phone call this very morning with people who were adamant that mixing the two was bad, so other people may have different experiences. Due to limited time, I decided not to press them to see if they were reporting bitter experience of days past, or whether they actually know it to still be a problem now that the build environment has been normalized between those two package sources. But strictly basing on my own experience, I vote for closing this issue as no longer being relevant for its original purpose.

As for the larger issue of wanting to detect environment issues in general, I think `pyct report` covers the reporting aspect, i.e. at least revealing what combinations of packages are in use, which is important because very often people will have no idea what combinations of package sources are involved. 

And for the more ambitious goal of automatically detecting cases we know are problematic, I can't currently think of any "known bad" configurations, e.g. known binary incompatibilities.  So I think all of that points to "close".
